### ISSUE TITLE
Code Mirror 6 support to function with Live Preview vim

### ISSUE THREAD

**Issue by 23727** [nadavspi/obsidian-relative-line-numbers#1]

**Comment by Philipp0205** [comment]
That would be awesome!

**Comment by Philipp0205** [comment]
With the help of @NothingIsLost I was able to get relative line numbers working again. 
I pushed a working plugin in a new branch [here, into the "cm6-support" branch.](https://github.com/Philipp0205/obsidian-relative-line-numbers/tree/cm6-support).

In the long run a PR to this repo would make sense but the plugin should support the legacy mode as well as the preview mode.

**Comment by Palvehelmed** [comment]
Very appreciate for your work!

**Comment by jdluque** [comment]
@Philipp0205 Thanks for your work with the workaround! I am using the plugin but the line number for the line the cursor is on reads 0 instead of the actual line number. Any idea how to fix this or why I am running into this problem?

@Palvehelmed Aside from the issue mentioned above, the plugin is working for me on v0.13.19.

**Comment by Philipp0205** [comment]
@Palvehelmed 
I can look into that tomorrow. 

Did you try to cone that repo into the obsidian plugins folder and run `npm run build`?

**Comment by jdluque** [comment]
@Philipp0205 Ah I see. I have taken a look at the link you shared. I don't follow all the code as I'm very new to ts and obsidian plugins but it seems the fix is as simple changing `src/index.ts:16` to `return lineNo.toString();`.

**Comment by jsjoeio** [comment]
Wow, this is funny. So I'm an Obsidian user, also using this plugin in my Obsidian setup. I came here to report that the plugin doesn't seem to be working in the latest version of Obsidian.

I'm also the author of that CM6 plugin. Thank you for raising a PR! I will merge it this evening and publish a new release.

**Comment by jsjoeio** [comment]
New version is out: https://www.npmjs.com/package/codemirror-line-numbers-relative/v/0.19.3-beta.2

**Comment by Philipp0205** [comment]
You guys are awesome! 

I updated the plugin version on [my fork](https://github.com/Philipp0205/obsidian-relative-line-numbers/tree/cm6-support) on the cm6-support branch as intermediate solution.

**Comment by Zeta611** [comment]
> You guys are awesome!
> 
> I updated the plugin version in [my fork](https://github.com/Philipp0205/obsidian-relative-line-numbers/tree/cm6-support) on the cm6-support branch as intermediate solution.

As a non-web developer, installation was not immediately obvious for me.
What I needed to do was:
1. Get the repo in the plugins directory.
2. Install `obsidian` and `@rollup/plugin-typescript`:
    ```sh
    npm install obsidian @rollup/plugin-typescript
    ```
3. Build:
    ```sh
    npm run build
    ```
4. Voila!

**Comment by jsjoeio** [comment]
I have a [working prototype](https://codesandbox.io/s/cm6-relative-line-numbers-v2-0gdcu) with two gutters! I assume it should _just work_ with the vim cm6 plugin too. I'm going to clean up the code and open a PR in my package. I'll let you know when a new release is out.

![relative-v2](https://user-images.githubusercontent.com/3806031/148650777-f1501180-9306-4490-9138-fcbd67391db4.gif)

**Comment by jsjoeio** [comment]
Published a new version with two gutters: https://www.npmjs.com/package/codemirror-line-numbers-relative/v/0.19.3-beta.3

**Comment by Philipp0205** [comment]
Thanks I appreciate it. Is it a thing to have the line numbers like this? I only know it with one gutter and the current line is the absolute number.

**Comment by jsjoeio** [comment]
Yes! I haven't checked in regular vim but VSCode does it this way. See here: https://github.com/jsjoeio/codemirror-line-numbers-relative/issues/1#issuecomment-1007819770

**Comment by free-150** [comment]
When will the new version be released

**Comment by Philipp0205** [comment]
That's great to hear!

**Comment by nadavspi** [comment]
@Philipp0205 I'm looking at your [branch](https://github.com/Philipp0205/obsidian-relative-line-numbers/blob/cm6-support/main.ts#L6) with the current version of `codemirror-line-numbers-relative` (`0.19.3-beta.4`). Does it work for you? I'm getting an error: `Uncaught (in promise) Error: Unrecognized extension value in extension set ([object Object]). This sometimes happens because multiple instances of @codemirror/state are loaded, breaking instanceof checks.`
`

**Comment by Philipp0205** [comment]
It worked for me but wehen I cloned my branch to replicate your error I get an differnt error now: 

`Plugin failure: obsidian-relative-line-numbers Error: ENOENT: no such file or directory, open '/home/philipp/Documents/2_Resources_part/Obsidian/Obsidian/.obsidian/plugins/obsidian-relative-line-numbers/main.js'

That is weird because in the mainifest the main.ts is defined. 
I will try to find out what it is.

**Comment by Philipp0205** [comment]
`npm install `and `npm run build` fixed it for me. Can you confirm that?

**Comment by nadavspi** [comment]
Published version [2.0.1](https://github.com/nadavspi/obsidian-relative-line-numbers/releases/tag/2.0.1) which should add support for Live Preview. Thanks for the CM plugin, @jsjoeio! I should have another release soon to fix the minor bug that I mention in the release, but figured it'd be good to get this out there.
